src/Core/Form/IdentifiableObject/Builder/FormBuilder.php

```diff
@@ -102,7 +102,7 @@ public function getForm(array $data = [], array $options = [])
         $defaultData = $this->dataProvider->getDefaultData();
         // Merge initial data in default data
         if (is_array($defaultData)) {
-            $data = array_merge($defaultData, $data);
+            $data = array_merge_recursive($defaultData, $data);
```

Comment:
ping @jolelievre Im pretty sure its a breaking change, but else it overrides all the FormDataProvider data if its multidimensional array (e.g. when I provide ['conditions']['customer'][] = 'some data' in controller, then whole 'conditions' tab is missing all the data from data provider and only contains that customer data.

Reply:
and now Im pretty sure AddressController is failing because of this :sweat: Might be other failures that are not tested, do you think there are any alternatives for this?